src/dataframe/dataframe.jl

```diff
@@ -1194,14 +1201,20 @@ function Base.append!(df1::DataFrame, df2::AbstractDataFrame)
     end
     ncol(df2) == 0 && return df1
 
-    _names(df1) == _names(df2) || throw(ArgumentError("Column names do not match"))
+    if cols == :orderequal && _names(df1) != _names(df2)
+        throw(ArgumentError("Column names do not match"))
```

Comment:
It would be nice to print different errors depending on whether column names differ (and in that case print the result of `symdiff`) or whether only the order differs. For the former, the next `if` could be used for both `:orderequal` and `:setequal`.

Reply:
Things are more complex than that. I have created more verbose error messages now.